### PR TITLE
This fixes #19

### LINKED ISSUE #19: form is loaded in $dirty state when used with moment object
Hello,

Here is the plunker forked from one of the example provided here:
[http://plnkr.co/edit/YcPQNstLRsFxND4r3a7Z](http://plnkr.co/edit/YcPQNstLRsFxND4r3a7Z)

I added a line showing the dirty state of the form at the very bottom.

This is a serious issue for me. Is this something to your knowledge? any workaround or fix for this?
Thank you.

### PR DESCRIPTION
**Pull request by shustariov-andrey**
Basically:
1) not setting undefined value into the model in $render 
2) not setting non-moment value in datepicker

### CODE REVIEW

File: dist/angular-eonasdan-datetimepicker.js [hunk 1]
```diff
@@ -25,12 +25,12 @@
                     }, true);
 
                     ngModel.$render = function () {
-                        if (!ngModel.$viewValue) {
-                            ngModel.$setViewValue(null);
-                        } else if (!moment.isMoment(ngModel.$viewValue)) {
+                        if (!_.isUndefined(ngModel.$viewValue) && !moment.isMoment(ngModel.$viewValue)) {
```
Comment (atais) [inline dist/angular-eonasdan-datetimepicker.js]: Since this project does not require `lodash` I would stick to `ngModel.$viewValue !== undefined`

**Comment by atais** [comment]
The problem this change introduces is though that you are unable to clear the time.

Check `single.html` example and click `Clear the time` button. It does not work anymore.

------
I had not written a **developer guide** but:

- Run `npm install` in main folder
- Run `gulp` (it will listen for changes)

Now check if all the options in `/examples`/ work fine.
**This should be automatic but I did not have time to play and write proper tests.**

**Comment by shustariov-andrey** [comment]
@atais, check now.

**Comment by atais** [comment]
Testing `single.html`

If you clear the **Input element** with bin button in the datetime-picker the **Input-group div element** is not cleared. It is working with master.

It does not work the other way round as well (clearing **Input-group div element** --> the **Input element** is not cleared).

:cry: 

I think I will start writing some automatic tests soon...

**Comment by atais** [comment]
@shustariov-andrey please update your fork with travis setup and tests :champagne:

**Comment by atais** [comment]
The tests are failing. Run them locally with `npm test`.

They fail also on travis, but the status code is 0 in the end...
But see the logs to see that it is not OK: https://travis-ci.org/atais/angular-eonasdan-datetimepicker/builds/197568101

**Comment by shustariov-andrey** [comment]
Yes, I can see that. I understand the problem. Will try to fix on this week-end.

**Comment by atais** [comment]
I have updated the `gulpfile.js`. Please merge it so I could see if the

**Comment by atais** [comment]
@shustariov-andrey your work motivated me to improve this project a lot and I wanted to say thank you. Yesterday, with the tests in place I have figured out the way to handle the issue properly and I have already coded the solution in PR #21 .

Thanks for your work and you can expect a fixed version out soon!

**Comment by shustariov-andrey** [comment]
@atais Glad to hear, will start to use it)

**Pull request closed without merging**